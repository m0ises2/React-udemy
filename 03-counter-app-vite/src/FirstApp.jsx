import PropTypes from "prop-types";

export const FirstApp = ( { title, subtitle, age } ) => {
    return (
        <>
            <h1 data-testid='test title'> { title } </h1>
            <p data-testid='test subtitle'> { subtitle } </p>
            <p> { age } years old </p>
        </>
    ) // Esto es un fragmento
}

FirstApp.propTypes = {
    age: PropTypes.number.isRequired,
    subtitle: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired
};

FirstApp.defaultProps = {
    age: 35,
    subtitle: 'Subtitulo por defecto'
};
