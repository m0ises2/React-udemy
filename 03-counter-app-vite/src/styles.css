html, body {
    background-color: rgb(0, 0, 0);
    color: white;
    font-family: Arial, Helvetica, sans-serif;
    font-size: 1.3rem;
    padding: 70px;
}

button {
    padding: 15px;
    font-size: 1.3rem;
}