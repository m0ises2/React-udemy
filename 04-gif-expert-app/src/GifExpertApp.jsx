import { useState } from 'react';
import { AddCategory, GifGrid } from './components';

export const GifExpertApp = () => {
  const [categories, setCategories] = useState([]);
  
  const onAddCategory = (newCategory) => {
    if (categories.includes(newCategory.toLowerCase())) return;
    
    setCategories([newCategory.toLowerCase(),...categories]);
  };

  return (
    <>
      {/* Titulo */}
      <h1>GitExpertApp</h1>

      {/* Input */}
      <AddCategory 
        onNewCategory={ onAddCategory }
      />
      
      {/* Lista de Gif */}
      { 
        categories.map(category => (
          <GifGrid 
            key={ category }
            category={ category }
          />
        ))
      }
        {/* Gif Item */}

    </>
  );
};
