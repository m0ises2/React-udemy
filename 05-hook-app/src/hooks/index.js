export * from './UseCounter';
export * from './useFetch';
export * from './useForm';