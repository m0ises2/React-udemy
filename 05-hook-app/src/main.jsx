import React from 'react';
import ReactDOM from 'react-dom/client';
// import { HooksApp } from "./HooksApp";
import './index.css';
// import { CounterApp } from './01-useState/CounterApp';
//import { CounterWithCustomHook } from './01-useState/CounterWithCustomHook';
// import { SimpleForm } from './02-useEffect/SimpleForm';
import { SimpleFormWithCustomHook } from './02-useEffect/SimpleFormWithCustomHook';

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>

    <SimpleFormWithCustomHook />

  </React.StrictMode>
);
