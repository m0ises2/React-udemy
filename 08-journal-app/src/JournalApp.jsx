import { RouterProvider } from 'react-router-dom';

import { AppRouter } from './router/AppRouter';

export const JournalApp = () => {
  return (<RouterProvider router={AppRouter} />);
}
