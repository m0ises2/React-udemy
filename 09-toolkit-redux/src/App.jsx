import { useSelector, useDispatch } from 'react-redux'
import { increment, incrementBy, decrement} from './store/slices/counter/counterSlice'

import reactLogo from './assets/react.svg'
import viteLogo from '/vite.svg'
import './App.css'

function App() {
  const { counter } = useSelector((state) => state.counter)
  const dispatch = useDispatch()

  return (
    <>
      <div>
        <a href="https://vitejs.dev" target="_blank">
          <img src={viteLogo} className="logo" alt="Vite logo" />
        </a>
        <a href="https://react.dev" target="_blank">
          <img src={reactLogo} className="logo react" alt="React logo" />
        </a>
      </div>
      <p>count is { counter }</p>
      <div className="card">
        <button onClick={() => dispatch(increment())}>
          Increment
        </button>
        <button onClick={() => dispatch(decrement())}>
          Decrement
        </button>
        <button onClick={() => dispatch(incrementBy(3))}>
          Increment by 3
        </button>
      </div>
    </>
  )
}

export default App
