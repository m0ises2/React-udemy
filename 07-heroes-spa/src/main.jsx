import React from 'react';
import ReactDOM from 'react-dom/client';
import './styles.css';
import { AppRouter } from './router/AppRouter';
import { AuthProvider } from './auth';

ReactDOM.createRoot(document.getElementById('root')).render(
  <AuthProvider>
    
    <AppRouter />

  </AuthProvider>,
)
