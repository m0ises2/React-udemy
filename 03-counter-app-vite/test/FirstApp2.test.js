import { render, screen } from '@testing-library/react';
import { FirstApp } from "../src/FirstApp";

describe('Pruebas en FirstApp', () => {
    const title = 'Hola, soy Moises';
    const subtitle = 'Soy un subtitle';

    test('Debe hacer match con el snapshot', () => {
        const { container } = render(<FirstApp title={title} />);
        expect(container).toMatchSnapshot();
    });

    test('Debe mostrar el mensaje "Hola, soy Moises"', () => {
        render(<FirstApp title={title} />);

        expect(screen.getByTestId('test title')).toBeTruthy();
    });

    test('Debe mostrar el titulo en un h1', () => {
        render(<FirstApp title={title} />);
        
        expect(screen.getByRole('heading', { level: 1 }).innerHTML).toContain(title);
    });

    test('Debe mostrar el subtitulo enviado por props', () => {
        render(<FirstApp title={title} subtitle={subtitle} />);

        expect(screen.getAllByText(subtitle).length).toBe(1);
    })
 });
