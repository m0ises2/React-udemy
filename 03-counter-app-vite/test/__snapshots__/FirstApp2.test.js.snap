// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en FirstApp Debe hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test title"
  >
     
    Hola, soy Moises
     
  </h1>
  <p
    data-testid="test subtitle"
  >
     
    Subtitulo por defecto
     
  </p>
  <p>
     
    35
     years old 
  </p>
</div>
`;
